src/Fantomas.Tests/NumberOfItemsInfixOperatorExpressionTests.fs

```diff
@@ -0,0 +1,277 @@
+    formatSourceString false """[10 .. -1 .. 1]""" config
+    |> should equal """[ 10 .. -1 .. 1 ]
+"""
+
+[<Test>]
+let ``long expression with pipe should be single line`` () =
+    formatSourceString false "let a = List.init 40 (fun i -> generateThing i a) |> List.map mapThingToOtherThing" config
+    |> prepend newline
+    |> should equal """
+let a =
+    List.init 40 (fun i -> generateThing i a) |> List.map mapThingToOtherThing
+"""
+
+[<Test>]
+let ``modulo operator on same line, 780`` () =
+    formatSourceString false """let hasUnEvenAmount regex line = (Regex.Matches(line, regex).Count - Regex.Matches(line, "\\\\" + regex).Count) % 2 = 1
+"""  config
+    |> prepend newline
+    |> should equal """
+let hasUnEvenAmount regex line =
+    (Regex.Matches(line, regex).Count - Regex.Matches(line, "\\\\" + regex).Count) % 2 = 1
+"""
+
+[<Test>]
+let ``parameter after multiline string, 783`` () =
+    formatSourceString false "
+let ``match bang`` () =
+    formatSourceString false \"\"\"
+async {
+    match! myAsyncFunction() with
+    | Some x -> printfn \"%A\" x
+    | None -> printfn \"Function returned None!\"
+}\"\"\"   config
+    |> prepend newline
+    |> should equal \"\"\"
+async {
+    match! myAsyncFunction () with
+    | Some x -> printfn \"%A\" x
+    | None -> printfn \"Function returned None!\"
+}
+\"\"\"
+"    config
+    |> prepend newline
+    |> should equal "
+let ``match bang`` () =
+    formatSourceString false \"\"\"
+async {
+    match! myAsyncFunction() with
+    | Some x -> printfn \"%A\" x
+    | None -> printfn \"Function returned None!\"
+}\"\"\" config
+    |> prepend newline
+    |> should equal \"\"\"
+async {
+    match! myAsyncFunction () with
+    | Some x -> printfn \"%A\" x
+    | None -> printfn \"Function returned None!\"
+}
+\"\"\"
+"
+
+[<Test>]
+let ``combining lines breaks function precedence 488`` () =
+    formatSourceString false """fun () -> ()
+|> Some
+"""  config
+    |> prepend newline
+    |> should equal """
+fun () -> ()
+|> Some
+"""
+
+[<Test>]
+let ``function with LPAREN_STAR_RPAREN`` () =
+    formatSourceString false """
+let private distanceBetweenTwoPoints (latA, lngA) (latB, lngB) =
+    if latA = latB && lngA = lngB then
+        0.
+    else
+        let theta = lngA - lngB
+
+        let dist =
+            Math.Sin(deg2rad (latA))
+            * Math.Sin(deg2rad (latB))
+            + (Math.Cos(deg2rad (latA))
+               * Math.Cos(deg2rad (latB))
+               * Math.Cos(deg2rad (theta)))
+            |> Math.Acos
+            |> rad2deg
+            |> (*) (60. * 1.1515 * 1.609344)
+
+        dist
+"""  config
+    |> prepend newline
+    |> should equal """
+let private distanceBetweenTwoPoints (latA, lngA) (latB, lngB) =
+    if latA = latB && lngA = lngB then
+        0.
+    else
+        let theta = lngA - lngB
+
+        let dist =
+            Math.Sin(deg2rad (latA)) * Math.Sin(deg2rad (latB))
```

Comment:
This doesn't seem correct to me, `Math.Sin(deg2rad (latA)) * Math.Sin(deg2rad (latB))` does not have any newline infix involved and is over 50 characters long.
![image](https://user-images.githubusercontent.com/2621499/95651400-76ff4d00-0aea-11eb-822f-8264d4e22bb9.png)